src/rte/mo_rte_sw.jl

```diff
@@ -28,105 +28,131 @@
+#    integer :: icol
 
-    real(wp), dimension(:,:,:), allocatable :: gpt_flux_up, gpt_flux_dn, gpt_flux_dir
-    real(wp), dimension(:,:),   allocatable :: sfc_alb_dir_gpt, sfc_alb_dif_gpt
+#    real(wp), dimension(:,:,:), allocatable :: gpt_flux_up, gpt_flux_dn, gpt_flux_dir
+#    real(wp), dimension(:,:),   allocatable :: sfc_alb_dir_gpt, sfc_alb_dif_gpt
     # ------------------------------------------------------------------------------------
-    ncol  = atmos%get_ncol()
-    nlay  = atmos%get_nlay()
-    ngpt  = atmos%get_ngpt()
-    nband = atmos%get_nband()
+    wp = eltype(mu0)
+    wl = eltype(top_at_1)
+
+    ncol  = get_ncol(atmos)
+    nlay  = get_nlay(atmos)
+    ngpt  = get_ngpt(atmos)
+    nband = get_nband(atmos)
     error_msg = ""
 
     # ------------------------------------------------------------------------------------
     #
     # Error checking -- consistency of sizes and validity of values
     #
     # --------------------------------
-    if(.not. fluxes%are_desired()) then
+    if(!are_desired(fluxes)) 
       error_msg = "rte_sw: no space allocated for fluxes"
       return
-    end if
+    end
 
     #
     # Sizes and values of input arrays
     #
-    if(     size(mu0, 1)                                /=  ncol        ) &
+    if( size(mu0)[1] !=  ncol )
       error_msg = "rte_sw: mu0 inconsistently sized"
-    if(any_vals_outside(mu0, 0._wp, 1._wp)) &
+    end
+
+    if(any_vals_outside(mu0, wp(0), wp(1))) 
       error_msg = "rte_sw: one or more mu0 <= 0 or > 1"
+    end
 
-    if(any([size(inc_flux, 1),    size(inc_flux, 2)]    /= [ncol, ngpt])) &
+    if(any([size(inc_flux)[1], size(inc_flux)[2]] != [ncol, ngpt])) 
       error_msg = "rte_sw: inc_flux inconsistently sized"
-    if(any_vals_less_than(inc_flux, 0._wp)) &
+    end
+
+    if(any_vals_less_than(inc_flux, wp(0))) 
       error_msg = "rte_sw: one or more inc_flux < 0"
-    if(present(inc_flux_dif)) then
-      if(any([size(inc_flux_dif, 1),    size(inc_flux_dif, 2)]    /= [ncol, ngpt])) &
+    end
+
+    if(present(inc_flux_dif))
+      if(any([size(inc_flux_dif)[1], size(inc_flux_dif)[2]] != [ncol, ngpt]))
         error_msg = "rte_sw: inc_flux_dif inconsistently sized"
-      if(any_vals_less_than(inc_flux_dif, 0._wp)) &
+      end
+      if(any_vals_less_than(inc_flux_dif, wp(0)))
         error_msg = "rte_sw: one or more inc_flux_dif < 0"
-    end if
+      end
+    end
 
-    if(any([size(sfc_alb_dir, 1), size(sfc_alb_dir, 2)] /= [nband, ncol])) &
+    if(any([size(sfc_alb_dir)[1], size(sfc_alb_dir)[2]] != [nband, ncol])) 
       error_msg = "rte_sw: sfc_alb_dir inconsistently sized"
-    if(any_vals_outside(sfc_alb_dir,  0._wp, 1._wp)) &
+    end
+    if(any_vals_outside(sfc_alb_dir,  wp(0), wp(1)))
       error_msg = "rte_sw: sfc_alb_dir out of bounds [0,1]"
-    if(any([size(sfc_alb_dif, 1), size(sfc_alb_dif, 2)] /= [nband, ncol])) &
+    end
+    if(any([size(sfc_alb_dif)[1], size(sfc_alb_dif)[2]] != [nband, ncol])) 
```

Comment:
```suggestion
    if(any([size(sfc_alb_dif)[1], size(sfc_alb_dif)[2]] .!= [nband, ncol])) 
```
These are vector operations, so we need to broadcast `!=` by putting `.` before it.